libctru/source/font.c

```diff
@@ -26,17 +25,34 @@ Result fontEnsureMapped(void)
 		return res;
 
 	g_sharedFont = (CFNT_s*)(sharedFontAddr+0x80);
-	charPerSheet = g_sharedFont->finf.tglp->nRows * g_sharedFont->finf.tglp->nLines;
 	return 0;
 }
 
-int fontGlyphIndexFromCodePoint(u32 codePoint)
+void fontFixPointers(CFNT_s* font)
 {
-	int ret = g_sharedFont->finf.alterCharIndex;
+	font->finf.tglp = (TGLP_s*)((u32)(font->finf.tglp) + (u32) font);
+	font->finf.tglp->sheetData = (u8*)((u32)(font->finf.tglp->sheetData) + (u32) font);
+
+	font->finf.cmap = (CMAP_s*)((u32)(font->finf.cmap) + (u32) font);
+	CMAP_s* cmap = font->finf.cmap;
+	for (; cmap->next; cmap = cmap->next)
+		cmap->next = (CMAP_s*)((u32)(cmap->next) + (u32) font);
+
+	font->finf.cwdh = (CWDH_s*)((u32)(font->finf.cwdh) + (u32) font);
+	CWDH_s* cwdh;
+	for (cwdh = font->finf.cwdh; cwdh->next; cwdh = cwdh->next)
+		cwdh->next = (CWDH_s*)((u32)(cwdh->next) + (u32) font);
+}
+
+int fontGlyphIndexFromCodePoint(CFNT_s* font, u32 codePoint)
+{
+	if (!font)
+		font = g_sharedFont;
+	int ret = font->finf.alterCharIndex;
```

Comment:
Add another `font` null check here (can happen if g_sharedFont is null) and return -1 in that case.